src/Workspaces/Core/Portable/CodeActions/CodeAction_Cleanup.cs

```diff
@@ -89,7 +94,9 @@ internal static async Task<Solution> CleanSyntaxAndSemanticsAsync(
             using var _ = ArrayBuilder<(DocumentId documentId, CodeCleanupOptions options)>.GetInstance(documentIds.Length, out var documentIdsAndOptions);
             foreach (var documentId in documentIds)
             {
-                var document = changedSolution.GetRequiredDocument(documentId);
+                var document = documentId.IsSourceGenerated
+                    ? changedSolution.GetRequiredSourceGeneratedDocumentForAlreadyGeneratedId(documentId)
+                    : changedSolution.GetRequiredDocument(documentId);
```

Comment:
in this case, i wouldn't mind an optiona lparameter to GetRequiredDocument stating it wants thsi behavior.  the default behavior should be to NOT get this i don't think.  but jason may feel differently.

Reply:
An optional parameter will certainly make the consumption easier, so I'll go ahead and do that. It doesn't prevent us needing to find each individual spot where we need to make the change though, so I would still argue for it to be the default behaviour, even if only for Razor source generated documents.